pycbc/population/scale_injections.py

```diff
@@ -0,0 +1,796 @@
+                
+                all_idx = np.array([np.array_equal(inj_d, uu) for uu in unique_distr]).astype(float)
+                idx = np.where(all_idx == 1)[0][0]
+                p_in += prob_mass[idx] * prob_dist[idx] * prob_spin * J * (1 + z_inj)**2 
+    
+            p_out_in = p_out/p_in
+            i_inj += np.sum(p_out_in)
+            i_det += np.array([np.sum((p_out_in)[np.where(data[thr_var][i] > thr)]) for thr in thr_val])
+            i_det_sq += np.array([np.sum((p_out_in)[np.where(data[thr_var][i] > thr)]**2) for thr in thr_val])
+            
+            idx_snr = np.where(data['stat'][i] > 8.0)
+            snrs = data['stat'][i][idx_snr]
+            ratios = p_out_in[idx_snr]/max(p_out_in[idx_snr])  
+            rndn = np.random.uniform(0, 1, len(ratios))
+            idx_ratio = np.where(ratios > rndn)                              
+            snr_falloff.append(snrs[idx_ratio])
+            
+        inj_V0 = injection_chunks[key]['inj_astro_vol']
+        injection_chunks[key]['ninj'] = i_inj
+        injection_chunks[key]['ndet'] = i_det
+        injection_chunks[key]['ndetsq'] = i_det_sq
+        injection_chunks[key]['VT'] = ((inj_V0*i_det/i_inj) * (data['gps'][1] - data['gps'][0])/31557600)
+        injection_chunks[key]['VT_err'] = np.sqrt(i_det_sq)/i_det
+        injection_chunks[key]['snr_falloff'] = np.hstack(np.array(snr_falloff).flat)
+    
+    return injection_chunks
+
+def process_injections(hdffile, inj_type = 'pipeline'):
+    """Function to read in the injection file and extract the found injections and all injections
+
+       Parameters
+       ----------
+       hdffile: hdf file
+           File for which injections are to be processed
+
+       Returns
+       -------
+       data: dictionary
+           Dictionary containing injection read from the input file
+    """
+    data = {}
+    
+    with h5py.File(hdffile, 'r') as inp:
+        found_index = inp['found_after_vetoes/injection_index'][:]
+
+        for param in _save_params:
+            data[param] = inp['injections/'+param][:]
+        
+        ifar = np.zeros_like(data[_save_params[0]])
+        ifar[found_index] = inp['found_after_vetoes/ifar'][:]
+        
+        data['ifar'] = ifar
+        
+        stat = np.zeros_like(data[_save_params[0]])
+        stat[found_index] = inp['found_after_vetoes/stat'][:]
+        
+        data['stat'] = stat
+
+    return data
+
+def merge_injections(all_inj):
+    """Merge injections across chunks.
+
+       Parameters
+       ----------
+       all_inj: dictionary
+           dictionary containing injections for various chunks
+
+       Returns
+       -------
+       data: dictionary
+           Dictionary containing the merged injections
+    """
+    injs = {}
+    for data in np.append(_save_params, 'found'):
+        injs[data] = np.concatenate([inj[data] for inj in tuple(all_inj.values())])
+   
+    return injs
+        
+def m1m2_to_mcheta(m1, m2):
```

Comment:
@vaibhavtewari Can you replace these types of functions with the equivelant in pycbc.conversions? Or if you have some new ones, please make a PR with just that change and add it to the pycbc.conversions module.

Reply:
I am just using m1m2->mch conversion. Now using from pycbc.conversion